### PR TITLE
DAOS-2793 dfuse: update readdir to support an out of order offset

### PR DESCRIPTION
**Pull request by mchaarawi**
- this supports the telldir/seekdir use case
- also fix some enumeration bugs in DFS

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/dfuse/ops/readdir.c [hunk 1]
```diff
@@ -166,23 +170,58 @@ dfuse_cb_readdir(fuse_req_t req, struct dfuse_inode_entry *inode,
...
-		 * returned from the last call on the OH, otherwise the offset
-		 * was corrupted.
+		 * otherwise we are starting at an earlier offset where we left
+		 * off on last readdir, so restart by first enumerating that
+		 * many entries. This is the telldir/seekdir use case.
 		 */
-		if (offset != (off_t)&oh->doh_anchor)
-			D_GOTO(err, rc = EIO);
+
+		memset(&oh->doh_anchor, 0, sizeof(oh->doh_anchor));
```
Comment (ashleypittman) [inline src/client/dfuse/ops/readdir.c]: Is there an initialiser for this?

Comment (mchaarawi) [reply src/client/dfuse/ops/readdir.c]: in the oh allocation in opendir

Comment (ashleypittman) [reply src/client/dfuse/ops/readdir.c]: I meant a daos_anchor_init() function or similar.

Comment (mchaarawi) [reply src/client/dfuse/ops/readdir.c]: ah, no there isn't.
Liang is considering updating the interface for the anchors.

**Pull request merged**